README.md

```diff
@@ -124,6 +124,34 @@ prettier --write 'src/**/*.js' 'bin/*.js'
 
 In the future we will have better support for formatting whole projects.
 
+#### Pre-commit hook for changed files
+
+[🚫💩 lint-staged](https://github.com/okonet/lint-staged) can re-format your files that are marked as "staged" via `git add`  before you commit.
+
+Install it along with [pre-commit](https://github.com/observing/pre-commit) (or [husky](https://github.com/typicode/husky)):
+
+```bash
+npm install --save-dev lint-staged pre-commit
+```
+
+and add this config to your `package.json`:
+
+```json
+{
+  "scripts": {
+    "lint:staged": "lint-staged"
+  },
+  "lint-staged": {
+    "*.js": [
+      "prettier --write",
+      "git add"
```

Comment:
Won't this also add any changes in the working dir that were not yet staged? So if I partially stage some files (or stage them and then fix another issue before committing), prettier will run on the version in the working dir and then stage that?

Should probably `git stash` before running prettier, and `git stash pop` after `git add`, no?